File: abc_dz_2/abc_dz2_source/classes/bus.cpp
//
// Created by Дмитрий Татьянченко on 18.10.2021.
//

#include "bus.h"

Bus::Bus(int fuel_tank_capacity, double fuel_consumption, short passenger_capacity) :
        Vehicle(fuel_tank_capacity, fuel_consumption) {

    this->type_ = (Vehicle::BUS);

    if (passenger_capacity < 0) {
        throw std::invalid_argument("speed limit < 0");
    }

    this->passenger_capacity_ = passenger_capacity;
}

std::string Bus::toString() {
    return "Bus: " + this->Vehicle::toString() +
           " passenger capacity: " + std::to_string(this->passenger_capacity_);
}

File: abc_dz_3/abc_dz3_source/vehicle/bus.py
from vehicle.vehicle import *


class Bus(Vehicle):
    def __init__(self, fuel_tank_capacity, fuel_consumption, passenger_capacity):
        super().__init__(fuel_tank_capacity, fuel_consumption)
        self.__passenger_capacity = passenger_capacity

    def to_string(self):
        return "BUS " + super().to_string() + " passenger capacity (ppl): " + str(self.__passenger_capacity)
